store/store.go

```diff
@@ -513,6 +556,23 @@ func LoadBlockStoreState(db dbm.DB) cmtstore.BlockStoreState {
 	return bsj
 }
 
+// LoadTxStatus retrieves the status of a transaction from the block store given its hash.
+func (bs *BlockStore) LoadTxStatus(txHash []byte) *cmtstore.TxStatus {
+	bz, err := bs.db.Get(calcTxHashKey(txHash))
+	if err != nil {
+		panic(err)
+	}
+	if len(bz) == 0 {
+		return nil
+	}
+
+	var txs cmtstore.TxStatus
+	if err = proto.Unmarshal(bz, &txs); err != nil {
+		panic(fmt.Errorf("unmarshal to TxIndex failed: %w", err))
```

Comment:
fwiw I like the `TxIndex` or `TxLocation` name better than `TxStatus` but I'm not strongly opinionated if we're eventually going to add more fields to `TxStatus` then that name prob makes more sense. Regardless, I think this instance failed to update after a name change:

```suggestion
		panic(fmt.Errorf("unmarshal to TxStatus failed: %w", err))
```